src/groovy/com/unifina/domain/CustomStreamIDValidator.groovy

```diff
@@ -11,12 +11,21 @@ class CustomStreamIDValidator {
 		boolean isOwnedBy(String domain, User owner)
 	}
 
+	// ENS domain rules:
+	// - must contain two or more segments separated by dot
+	// - segments can contain chars a-z, A-Z, 0-9 and -
+	// https://docs.ens.domains/contract-api-reference/name-processing
+	private static final String ENS_DOMAIN_REGEX = "(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z0-9-]+"
+
 	// path rules:
 	// - must start by slash
 	// - can contain chars a-z, A-Z, 0-9 and -_.
 	// - must not end with non-word character (in this case - or .)
 	// - can have segments separated by slashes (two consecutive slashes is not allowed)
-	public static final Pattern REGEX = Pattern.compile("^((?:[\\w-]+\\.?)*\\w)/(?:[\\w\\.-]+/?)*\\w\$")
+	private static final String PATH_REGEX = "(?:(?:/[\\w\\.-]+)+\\w)|(?:/\\w)"
+
+	private static final String ETHEREUM_ADDRESS_REGEX = "0x[a-fA-F0-9]{40}"
```

Comment:
`{40}` limit is good! Any chance to get similar limit to ENS_DOMAIN_REGEX?

https://github.com/ethereum/EIPs/blob/master/EIPS/eip-137.md#name-syntax

Looks like the limit is 255 chars. I think the current database field's size is a limitation. Should that be migrated?

Reply:
The ENS domain regex is mostly for catching the most typical errors early. If it looks ok, we query the ENS registry, and that query won't return OK for invalid ENS names. 

But yes, we could test that the full streamId has max 255 chars. I'll fix.

Reply:
Fixed

Reply:
👍